### PR TITLE
Add Wordle-like game example

### PR DESCRIPTION
**Pull request by jacobdweightman**
This example probably still needs some work, and there may be IP issues related to the name. This is what an example game currently looks like:

Screenshot 2022-12-20 at 1 45 39 PM

### CODE REVIEW

File: wordle/methods/guest/src/bin/wordle.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#![no_main]
+// #![no_std]
+
+use risc0_zkvm_guest::{env, sha};
+use wordle_core::{WordFeedback, LetterFeedback, GameState};
+// use risc0_zkp::core::sha::Digest;
```
Comment (mars-risc0) [inline wordle/methods/guest/src/bin/wordle.rs:6]: if this is not necessary, maybe remove it?

File: wordle/host/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+use std::io;
+
+use methods::{WORDLE_ID, WORDLE_PATH};
+use risc0_zkvm::host::{Prover, Receipt};
+use crate::wordlist::words::pick_word;
+use risc0_zkvm::serde::{to_vec};
+// use wordle_core::{GameState};
```
Comment (mars-risc0) [inline wordle/host/src/main.rs:7]: perhaps remove this if it is no longer needed

File: wordle/host/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+    for _ in 0..6 {
+        let guess_word = read_stdin_guess();
+        let receipt = server.check_round(guess_word.as_str());
+        let score = player.check_receipt(receipt);
+        print_wordle_feedback(guess_word.as_str(), &score);
+        if score.iter().all(|x| *x == 0u32) {
+            println!("You win!");
+            break;
+        }
+    }
```
Comment (mars-risc0) [inline wordle/host/src/main.rs:96]: might be nice to print a message when the game ends and you don't win, too

**Review by jbruestle — APPROVED** [review]
Love it!  Super clear and cool use case.  Technically real Wordle only allows guesses that are in fact words, which this example doesn't check.  Such a check should probably be in the server but outside the ZKP (since the server saying 'invalid guess' doesn't reveal anything and can be independently verified by the client).  But we can also ignore that point, so I'm approving.

**Comment by choongng** [comment]
I love it. Is the valid word check too expensive to do inside? The "uses external blob" option is one of the things I've heard identified as never going to be feasible on-chain for Ethereum and also something that should work with our tools.

**Review by choongng — APPROVED** [review]

**Comment by jacobdweightman** [comment]
I think we can definitely iterate on some of the details here like checking that guesses are valid words, but I think it's approximately ready to land. I think the one big question is whether I need to scrub the name "wordle" from the code for legal reasons?

**Review by flaub — APPROVED** [review]

File: wordle/core/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,18 @@
+use risc0_zkp::core::sha::Digest;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub enum LetterFeedback {
+	LetterCorrect,
+	LetterPresent,
+    #[default]
+	LetterMiss,
```
Comment (flaub) [inline wordle/core/src/lib.rs:9]: Might want to run `cargo fmt` on this

**Comment by flaub** [comment]
Please add an entry to the CI: https://github.com/risc0/risc0-rust-examples/blob/main/.github/workflows/main.yml#L17

**Pull request merged**